ui/pages/DataManagement/reducers.js

```diff
@@ -54,7 +54,17 @@ export const uploadQcPipelineOutput = values => submitRequest(
 
 export const deleteEsIndex = index => submitRequest('delete_index', RECEIVE_ELASTICSEARCH_STATUS, { index })
 
-const loadMultipleData = (path, getUpdateData, dispatchType, formatSuccessMessage) => values => (dispatch) => {
+const executeMultipleRequests = (requests, onSuccess, warnings) => Promise.all(requests.map(
+  ([entityUrl, entityId, body]) => new HttpRequestHelper(
+    entityUrl,
+    onSuccess,
+    e => warnings.push(`Error loading ${entityId}: ${e.body && e.body.error ? e.body.error : e.message}`),
+  ).post(body),
+))
+
+const loadMultipleData = (
+  path, getUpdateData, dispatchType, formatSuccessMessage, maxConcurrentRequests = 50,
```

Comment:
could we add an assertion that `maxConcurrentRequests` is `> 1`?

Reply:
we currently have no js function argument validation anywhere in our code, so it seems pretty out of scope to add it here